### PR TITLE
openai realtime beta

### PR DESCRIPTION
**Pull request by kwindla**
Okay, I think this is ready for review.

Function calling has been tested pretty heavily. Most things are configurable.

I've wired up RTVI/configuration frames, though I have not tested much of that.

I also added a decent start on saving/loading context portably between LLMs. The idea is to standardize on the OpenAI messages format, converting to/from when we save/load. This is going be rife with corner cases, but I've tested openai, openai-realtime, and anthropic. Basic messages and function calling work robustly.

In `examples/foundational`

```
20a-persistent-context-openai.py
20b-persistent-context-openai-realtime.py
20c-persistent-context-anthropic.py
```

### CODE REVIEW

**Comment by mercuryyy** [comment]
Can you possibly include a demo using the web call demo as well ? - https://github.com/pipecat-ai/pipecat/tree/main/examples/websocket-server

File: src/pipecat/services/openai_realtime_beta/llm_and_context.py [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+            obj.__setup_local()
+        return obj
+
+    # todo
+    #   - finish implementing all frames
+    #   - add message conversion functions to OpenAILLMContext base class
+
+    def from_standard_message(self, message):
+        if message.get("role") == "assistant" and message.get("tool_calls"):
+            tc = m.get("tool_calls")[0]
```
Comment (joachimchauvet) [inline src/pipecat/services/openai_realtime_beta/llm_and_context.py:106]: Shouldn't that be `tc = message.get("tool_calls")[0]`?

File: src/pipecat/services/openai_realtime_beta/events.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
+import json
```
Comment (markbackman) [inline src/pipecat/services/openai_realtime_beta/events.py:1]: Add header with copyright and open source license

File: src/pipecat/services/openai_realtime_beta/llm_and_context.py [hunk 3]
```diff
@@ -0,0 +1,715 @@
+import asyncio
```
Comment (markbackman) [inline src/pipecat/services/openai_realtime_beta/llm_and_context.py:1]: Add header with copyright and open source license

**Review by markbackman — APPROVED** [review]
LGTM! Nice job 👏 

The only remaining thing is to add the copyright and license block to the top of the `llm_and_context.py` and `events.py` files.

**Pull request merged**